src/cephadm/tests/test_cephadm.py

```diff
@@ -2149,20 +2148,12 @@ def test_parse_yaml(self, cephadm_fs):
 '''
 
         cephadm_fs.create_file('spec.yml', contents=yaml)
-        retdic = [{'service_type': 'host', 'hostname': 'vm-00', 'addr': '192.168.122.44', 'labels': '- example1- example2'},
-                  {'service_type': 'host', 'hostname': 'vm-01', 'addr': '192.168.122.247', 'labels': '- grafana'},
-                  {'service_type': 'host', 'hostname': 'vm-02', 'addr': '192.168.122.165'},
-                  {'service_id': 'myrgw',
-                   'service_type': 'rgw',
-                   'spec':
-                   'rgw_frontend_ssl_certificate: |-----BEGIN PRIVATE '
-                   'KEY-----V2VyIGRhcyBsaWVzdCBpc3QgZG9vZi4gTG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQsIGNvbnNldGV0dXIgc2FkaXBzY2luZyBlbGl0ciwgc2VkIGRpYW0gbm9udW15IGVpcm1vZCB0ZW1wb3IgaW52aWR1bnQgdXQgbGFib3JlIGV0IGRvbG9yZSBtYWduYSBhbGlxdXlhbSBlcmF0LCBzZWQgZGlhbSB2b2x1cHR1YS4gQXQgdmVybyBlb3MgZXQgYWNjdXNhbSBldCBqdXN0byBkdW8=-----END '
-                   'PRIVATE KEY----------BEGIN '
-                   'CERTIFICATE-----V2VyIGRhcyBsaWVzdCBpc3QgZG9vZi4gTG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQsIGNvbnNldGV0dXIgc2FkaXBzY2luZyBlbGl0ciwgc2VkIGRpYW0gbm9udW15IGVpcm1vZCB0ZW1wb3IgaW52aWR1bnQgdXQgbGFib3JlIGV0IGRvbG9yZSBtYWduYSBhbGlxdXlhbSBlcmF0LCBzZWQgZGlhbSB2b2x1cHR1YS4gQXQgdmVybyBlb3MgZXQgYWNjdXNhbSBldCBqdXN0byBkdW8=-----END '
-                   'CERTIFICATE-----ssl: true'}]
```

Comment:
Is this UT not valid anymore? (just to make sure we are no deleting it by mistake)

Reply:
No, this test was for some of the more advanced yaml parsing it was trying before. That's why it has an rgw spec with a certificate and all. We were trying to manually parse yaml with even advanced fields like that. Part of the change here is removing that (because we don't need it) and having just parsing that works for host specs, since that's the only one we need to take action on (distributing the ssh keys) before passing the yaml to the cephadm mgr module. That's why the test was changed to only mention host stuff.

Reply:
Or, I guess I should say, I removed the non-host stuff from the output we check against but left the input the same. The idea is that specs that aren't hosts will get filtered out and then it will end up with just a set of hosts and addresses to which it needs to copy the ssh key. So the input to this test is still the same but it doesn't actually try to properly parse rgw specs or anything, just the host specs which have a simple, expected format.

Reply:
I see, fair enough meanwhile we are checking the hosts parsing functionality. The less "manual" parsing we try to do the better :)